bots/pr/src/main/java/org/openjdk/skara/bots/pr/LabelerWorkItem.java

```diff
@@ -71,69 +78,135 @@ private void updateLabelMessage(List<Comment> comments, List<String> newLabels)
+                message.append("\n");
+                message.append("</details>");
+            } else {
+                message.append("The following label");
+                if (newLabels.size() > 1) {
+                    message.append("s");
+                }
+                message.append(" will be automatically applied to this pull request:\n\n");
+                newLabels.stream()
+                        .sorted()
+                        .forEach(label -> message.append("- `" + label + "`\n"));
+                message.append("\n");
+                message.append("When this pull request is ready to be reviewed, an \"RFR\" email will be sent to the ");
+                message.append("corresponding mailing list");
+                if (newLabels.size() > 1) {
+                    message.append("s");
+                }
+                message.append(". If you would like to change these labels, use the ");
+                message.append("[/label](https://wiki.openjdk.org/display/SKARA/Pull+Request+Commands#PullRequestCommands-/label)");
+                message.append(" pull request command.");
             }
-            message.append(". If you would like to change these labels, use the ");
-            message.append("[/label](https://wiki.openjdk.org/display/SKARA/Pull+Request+Commands#PullRequestCommands-/label)");
-            message.append(" pull request command.");
+        } else {
+            message.append("A manual label command was issued before auto-labeling, so auto-labeling was skipped.");
         }
 
         message.append("\n");
         message.append(INITIAL_LABEL_MESSAGE);
+        message.append("\n");
+        message.append(String.format(LABEL_COMMIT_MARKER, commitHash));
         pr.addComment(message.toString());
     }
 
     @Override
     public Collection<WorkItem> prRun(ScratchArea scratchArea) {
-        if (bot.isAutoLabelled(pr)) {
+        // If no label configuration, return early
+        if (bot.labelConfiguration().allowed().isEmpty()) {
+            bot.setAutoLabelled(pr);
             return List.of();
         }
 
-        if (bot.labelConfiguration().allowed().isEmpty()) {
+        // If the label comment can be found, mark the pr as auto labelled
+        if (findComment(prComments(), INITIAL_LABEL_MESSAGE).isPresent()) {
             bot.setAutoLabelled(pr);
+        }
+
+        // Updating labels when new files are touched
+        if (bot.isAutoLabelled(pr)) {
+            try {
+                var oldLabels = new HashSet<>(pr.labelNames());
+                var newLabels = new HashSet<>(pr.labelNames());
+
+                var path = scratchArea.get(pr.repository());
+                var seedPath = bot.seedStorage().orElse(scratchArea.getSeeds());
+                var hostedRepositoryPool = new HostedRepositoryPool(seedPath);
+                var localRepo = PullRequestUtils.materialize(hostedRepositoryPool, pr, path);
```

Comment:
This is repeated further down, maybe worth breaking out into a method `materializeRepo`?

Reply:
Sure, will do.

Reply:
I found there is already a method `IntegrateCommand#materializeLocalRepo`